File: Greed/src/spacealert/core/planningPhase/commands/actionCards/RetrieveCardFromOwnActionBoardCommand.java
package spacealert.core.planningPhase.commands.actionCards;

import com.fasterxml.jackson.annotation.JsonProperty;
import spacealert.core.planningPhase.PlanningPhase;
import spacealert.core.planningPhase.Player;

import java.util.UUID;

public class RetrieveCardFromOwnActionBoardCommand implements IPlanningPhaseCommand {
    @JsonProperty
    private UUID cardId;

    public RetrieveCardFromOwnActionBoardCommand(UUID cardId) {
        this.cardId = cardId;
    }

    @Override
    public void execute(PlanningPhase planningPhase, Player player) {
        player.retrieveCardFromActionBoard(cardId);
    }
}

File: Greed/src/client/spacealert/src/models/commands/RetrieveCardFromOwnActionBoardCommand.ts
import { IsUUID } from "class-validator"

export default class RetrieveCardFromOwnActionBoardCommand {
  type = ".RetrieveCardFromOwnActionBoardCommand"

  @IsUUID()
  cardId: string

  constructor(cardId: string) {
    this.cardId = cardId
  }
}